pkg/lso/localvolumediscovery_test.go

```diff
@@ -0,0 +1,404 @@
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryCreate(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            string
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            "",
+		},
+		{
+			testLocalVolumeDiscovery: buildInValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            "localVolumeDiscovery 'name' cannot be empty",
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:            "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testLocalVolumeSetBuilder, err := testCase.testLocalVolumeDiscovery.Create()
+
+		if testCase.expectedError == "" {
+			assert.Equal(t, testLocalVolumeSetBuilder.Definition.Name, testLocalVolumeSetBuilder.Object.Name)
+			assert.Equal(t, testLocalVolumeSetBuilder.Definition.Namespace, testLocalVolumeSetBuilder.Object.Namespace)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryDelete(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            error
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            nil,
+		},
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(clients.GetTestClients(clients.TestClientParams{})),
+			expectedError:            fmt.Errorf("localVolumeDiscovery cannot be deleted because it does not exist"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		err := testCase.testLocalVolumeDiscovery.Delete()
+
+		if testCase.expectedError == nil {
+			assert.Nil(t, testCase.testLocalVolumeDiscovery.Object)
+			assert.Nil(t, err)
+		} else {
+			assert.Equal(t, testCase.expectedError.Error(), err.Error())
+		}
+	}
+}
+
+func TestLocalVolumeDiscoveryUpdate(t *testing.T) {
+	testCases := []struct {
+		testLocalVolumeDiscovery *LocalVolumeDiscoveryBuilder
+		expectedError            string
+		testNodeSelector         corev1.NodeSelector
+	}{
+		{
+			testLocalVolumeDiscovery: buildValidLVDObjectBuilder(buildLVDClientWithDummyObject()),
+			expectedError:            "",
+			testNodeSelector: corev1.NodeSelector{
+				NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+					MatchExpressions: []corev1.NodeSelectorRequirement{{
+						Key:      "cluster.ocs.openshift.io/openshift-storage",
+						Operator: "In",
+						Values:   []string{""},
+					}}},
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Nil(t, nil, testCase.testLocalVolumeDiscovery.Object)
+		testCase.testLocalVolumeDiscovery.WithNodeSelector(testCase.testNodeSelector)
+		_, err := testCase.testLocalVolumeDiscovery.Update()
+
+		if testCase.expectedError != "" {
```

Comment:
Check error unconditionally

Reply:
@trewest it isimpossible here as weel. The reason is different. For the Update method we always have a "conflict' error on demo client:
Message:"Operation cannot be fulfilled on localvolumediscoveries.local.storage.openshift.io \"auto-discover-devices\": object was modified", Reason:"Conflict"